### ISSUE TITLE
OAS - '$ref' properties on example values return invalid-ref errors.

### ISSUE THREAD

**Issue by kevinswiber** [stoplightio/spectral#1310]
When a `$ref` property exists in an example value, `@stoplight/json-ref-resolver` attempts to dereference the value.

My interpretation of the specification leads me to believe that this shouldn't happen.  From the OAS 3.0.3 spec:

Field Name | Type | Description
---|:---:|---
value | Any | Embedded literal example. The `value` field and `externalValue` field are mutually exclusive. To represent examples of media types that cannot naturally represented in JSON or YAML, use a string value to contain the example, escaping where necessary.

**To Reproduce**

1. Given this OpenAPI document:

```yaml
openapi: '3.0.0'
info:
  title: example-ref
  description: Incorrect validation on examples.
  version: '1.0'
  contact: {}
servers:
- url: http://localhost:3000
paths:
  /examples:
    post:
      description: Post an example.
      operationId: postExamples
      tags:
        - Example
      requestBody:
        description: An example body that contains a '$ref' property.
        content:
          'application/json':
            schema:
              type: object
              example:
                $ref: '#/should/not/validate'
      responses:
        '201':
          description: Created
tags:
  - name: Example
```

2. Run this CLI command:

```
spectral lint example.yaml
```

3. See error:

```
 23:23  error  invalid-ref  '#/should/not/validate' does not exist

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)
```

**Expected behavior**

Example values should be transparent to the OAS ruleset.  This is a valid OpenAPI definition.

**Environment (remove any that are not applicable):**

 - Library version: 5.5.0-beta9
 - OS: macOS Catalina 10.15.6

**Comment by slegarraga** [comment]
Agree with your reading — when `$ref` appears **inside an example value** (literal data), it should not be treated as a JSON Reference to resolve.

Worth checking whether the resolver runs on `example`/`examples` payloads before rules, or whether a specific rule walks example trees with ref semantics. A minimal repro like yours is perfect for a fix + regression test.

Happy to help validate if someone opens a PR.

**Comment by Patrycja-dz** [comment]
Reopening this - on first pass I grouped both issues under "$ref inside example data", but a closer look shows they describe different symptoms.

#2832 is about the error being reported at the wrong location after $ref resolution. The validation itself fires; only the reported path is wrong.

This issue is more fundamental - the resolver treats $ref inside a literal example value as a real JSON Reference and raises invalid-ref. The spec is valid; the error shouldn't appear at all.

Fixing this issue would likely fix #2832 as a side effect, but they are not the same bug.
